### ISSUE TITLE
Add delete() function to QueryWithSoftDelete

### ISSUE THREAD

**Issue by jforand** [miguelgrinberg/sqlalchemy-soft-delete#10]
Less of an issue and more of a feature request.

It would be cool if the QueryWithSoftDelete overrode the delete() function of BaseQuery so that rather than delete the object, it set the attribute `deleted ` of the model to true.

**Comment by miguelgrinberg** [comment]
I'm not sure if that's a good idea, because then you would need to jump through hoops if you ever need to actually delete an entry for real. It would also make the code look like it is doing a delete, versus the current option which makes it very clear that it is a soft delete.

**Comment by jforand** [comment]
Fair point. What about adding a soft_delete() function to the query instead?

**Comment by miguelgrinberg** [comment]
Yeah, a `soft_delete` seems reasonable. Would you like to attempt it and send a PR over?

**Comment by jforand** [comment]
https://github.com/miguelgrinberg/sqlalchemy-soft-delete/pull/11
